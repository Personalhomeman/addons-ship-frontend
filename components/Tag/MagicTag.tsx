import { TypeColors } from '@bitrise/bitkit';

import Tag from '.';

const colors: TypeColors[] = ['orange-3', 'yellow-4', 'green-3', 'blue-3', 'darkblue-3', 'violet-3'];

export type Props = {
  children: string;
  selected?: boolean;
  large?: boolean;
  onClick?: Function;
};

export const getStringSum = (s: string) => [...s.toLowerCase()].reduce((sum, char) => sum + char.charCodeAt(0), 0);

export default ({ children, ...props }: Props) => {
  const sum = getStringSum(children);
  const color = colors[sum % colors.length];

  return (
    <Tag color={color} textColor="white" {...props}>
      {children}
    </Tag>
  );
};
