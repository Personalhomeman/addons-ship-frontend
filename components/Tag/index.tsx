import cx from 'classnames';
import { TypeColors, Base, Text, variables } from '@bitrise/bitkit';

import css from './style.scss';

export type Props = {
  color: TypeColors;
  textColor: TypeColors;
  children: string;
  onClick?: Function;
  selected?: boolean;
  large?: boolean;
};

export default ({ color, textColor, children, selected, large, onClick }: Props) => (
  <Base
    backgroundColor={selected ? color : undefined}
    className={cx(css.Tag, { [css.clickable]: onClick })}
    paddingHorizontal={large ? 'x3' : 'x2'}
    paddingVertical={large ? 'x2' : 'x1'}
    borderRadius="x1"
    borderWidth={selected ? undefined : 'x1'}
    style={selected ? undefined : { borderColor: variables.colorMap[color] }}
    onClick={onClick}
  >
    <Text color={selected ? textColor : color} config={large ? '8' : '9'} uppercase>
      {children}
    </Text>
  </Base>
);
