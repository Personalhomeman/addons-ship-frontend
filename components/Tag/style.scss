.Tag {
  display: inline-block;
  cursor: pointer;

  &:hover {
    opacity: 0.8;
  }
}
