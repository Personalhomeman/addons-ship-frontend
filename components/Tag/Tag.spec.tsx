import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import Tag from '.';
import MagicTag, { getStringSum } from './MagicTag';

describe('Tag', () => {
  test('simple', () => {
    const tree = shallowToJson(
      shallow(
        <Tag color="blue-3" textColor="white">
          Some text
        </Tag>
      )
    );
    expect(tree).toMatchSnapshot();
  });

  test('selected', () => {
    const tree = shallowToJson(
      shallow(
        <Tag color="blue-3" textColor="white" selected>
          Some text
        </Tag>
      )
    );
    expect(tree).toMatchSnapshot();
  });

  test('large', () => {
    const tree = shallowToJson(
      shallow(
        <Tag color="blue-3" textColor="white" large>
          Some text
        </Tag>
      )
    );
    expect(tree).toMatchSnapshot();
  });
});

describe('MagicTag', () => {
  test('getStringSum', () => {
    expect(getStringSum('whatever')).toBe(870);
    expect(getStringSum('WhAtEvEr')).toBe(870);
    expect(getStringSum('sure thing')).toBe(1017);
  });

  test('simple', () => {
    const tree = shallowToJson(shallow(<MagicTag>magic âœ¨</MagicTag>));
    expect(tree).toMatchSnapshot();
  });
});
