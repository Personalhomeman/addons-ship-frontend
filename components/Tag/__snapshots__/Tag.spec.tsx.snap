// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MagicTag simple 1`] = `
<_default
  color="blue-3"
  textColor="white"
>
  magic âœ¨
</_default>
`;

exports[`Tag large 1`] = `
<Base
  borderRadius="x1"
  borderWidth="x1"
  className="Tag"
  paddingHorizontal="x3"
  paddingVertical="x2"
  style={
    Object {
      "borderColor": "#0D83CD",
    }
  }
>
  <Text
    color="blue-3"
    config="8"
    uppercase={true}
  >
    Some text
  </Text>
</Base>
`;

exports[`Tag selected 1`] = `
<Base
  backgroundColor="blue-3"
  borderRadius="x1"
  className="Tag"
  paddingHorizontal="x2"
  paddingVertical="x1"
>
  <Text
    color="white"
    config="9"
    uppercase={true}
  >
    Some text
  </Text>
</Base>
`;

exports[`Tag simple 1`] = `
<Base
  borderRadius="x1"
  borderWidth="x1"
  className="Tag"
  paddingHorizontal="x2"
  paddingVertical="x1"
  style={
    Object {
      "borderColor": "#0D83CD",
    }
  }
>
  <Text
    color="blue-3"
    config="9"
    uppercase={true}
  >
    Some text
  </Text>
</Base>
`;
